src/components/MultiHopTrade/components/TradeInput/TradeInput.tsx

```diff
@@ -547,10 +554,19 @@ export const TradeInput = ({ isCompact }: TradeInputProps) => {
   )
 
   const handleFormSubmit = useMemo(() => handleSubmit(onSubmit), [handleSubmit, onSubmit])
-  const handleTradeQuoteConfirm = useCallback(
-    () => (isUnsafeQuote ? setShouldShowWarningAcknowledgement(true) : handleFormSubmit()),
-    [handleFormSubmit, isUnsafeQuote],
-  )
+
+  // If the warning acknowledgement is shown, we need to handle the submit differently because we might want to show the streaming acknowledgement
+  const handleWarningAcknowledgementSubmit = useCallback(() => {
+    if (activeQuote?.isStreaming) return setShouldShowStreamingAcknowledgement(true)
+    handleFormSubmit()
```

Comment:
q: do we need this here as well?

Reply:
Absolutely, we need to submit after handling the acknowledgement if there are no more acknowledgement afterwhile